aspnetcore/security/blazor/webassembly/hosted-with-identity-server.md

```diff
@@ -223,6 +223,209 @@ The `LoginDisplay` component (*Shared/LoginDisplay.razor*) is rendered in the `M
 
 Run the app from the Server project. When using Visual Studio, select the Server project in **Solution Explorer** and select the **Run** button in the toolbar or start the app from the **Debug** menu.
 
+## Name and role claim with API authorization
+
+Identity Server can be configured to send `name` and `role` claims for authenticated users.
+
+> [!NOTE]
+> Currently, only one assigned role per user is supported.
+
+<!-- HOLD until the factory pattern issues are resolved
```

Comment:
What issue is being referred to here? Can it be linked?

Reply:
Just read the description and I see the issue you referred to here. Is that issue in the sample code or in Blazor?

Reply:
In Blazor I think. There are three possibilities ...

* `role` claim with JSON array because the user has multiple roles.

  ```
  "role": [ "role1", "role2" ]
  ```

  The factory works because the framework processes the array and populates `CustomUserAccount.Roles` (`string[]`) correctly. The factory makes a single `role` claim out of each of those. :+1:

* `role` claim isn't sent down by IS because the user doesn't have any roles. If `CustomUserAccount.Roles` (`string[]`) in the factory is `null`, the factory throws at the `foreach` loop. So to resolve this, I set the array initially in the custom user to an empty array ...

  ```csharp
  public class CustomUserAccount : RemoteUserAccount
  {
      [JsonPropertyName("role")]
      public string[] Roles { get; set; } = new string[0];
  }
  ```

  Now, `CustomUserAccount.Roles` is never `null`. Since there are no values in `CustomUserAccount.Roles` for a user without roles, the factory simply no-ops on setting any `role` claims, and that seems fine. :+1:

* `role` claim with a single string entry.

  ```
  "role": "role1"
  ```

  :boom: ... The factory code isn't reached. The framework fails to populate `CustomUserAccount.Roles` with a single entry for `role1`. Seems like it has to do with the property (`CustomUserAccount.Roles`) being an array with a string value received in the claim.